larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/TextAnalyzerModule.java

```diff
@@ -0,0 +1,141 @@
+package com.verygood.security.larky.modules.vgs.cerebro.text.analyzer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.analyzers.DefaultTextPIIAnalyzer;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.analyzers.NoopTextPIIAnalyzer;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.dto.TextPIIEntity;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.spi.TextPIIAnalyzer;
+import net.starlark.java.annot.Param;
+import net.starlark.java.annot.ParamType;
+import net.starlark.java.annot.StarlarkBuiltin;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.Starlark;
+import net.starlark.java.eval.StarlarkFloat;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.Set;
+
+@StarlarkBuiltin(
+    name = "TextPIIAnalyzer",
+    category = "BUILTIN",
+    doc = "Overridable Text PII Analysis API in Larky")
+public class TextAnalyzerModule implements TextPIIAnalyzer {
+
+  public static final TextAnalyzerModule INSTANCE = new TextAnalyzerModule();
+  public static final String ENABLE_INMEMORY_PROPERTY = "larky.modules.vgs.cerebro.piiAnalyzer.text.enableInMemory";
+
+  private TextPIIAnalyzer textPiiAnalyzer;
+
+  public TextAnalyzerModule() {
+
+    ServiceLoader<TextPIIAnalyzer> loader = ServiceLoader.load(TextPIIAnalyzer.class);
+    List<TextPIIAnalyzer> textAnalyzerProviders = ImmutableList.copyOf(loader.iterator());
+
+    if (Boolean.getBoolean(ENABLE_INMEMORY_PROPERTY)) {
+      textPiiAnalyzer = new DefaultTextPIIAnalyzer();
+    } else if (textAnalyzerProviders.isEmpty()) {
+      textPiiAnalyzer = new NoopTextPIIAnalyzer();
+    } else {
+      if (textAnalyzerProviders.size() != 1) {
+        throw new IllegalArgumentException(String.format(
+            "Cerebro expecting only 1 text PII data analyzer provider of type TextPiiAnalyzer, found %d",
+            textAnalyzerProviders.size()
+        ));
+      }
+
+      textPiiAnalyzer = textAnalyzerProviders.get(0);
+    }
+
+  }
+
+  @StarlarkMethod(
+      name = "analyze",
+      doc = "performs PII entities search, given supported language.",
+      parameters = {
+          @Param(
+              name = "text",
+              doc = "the text to analyze",
+              allowedTypes = {
+                  @ParamType(type = String.class),
+              }),
+          @Param(
+              name = "language",
+              doc = "two characters for the desired language in ISO_639-1 format.",
+              named = true,
+              defaultValue = "'EN'",
+              allowedTypes = {
+                  @ParamType(type = String.class),
+              }),
+          @Param(
```

Comment:
What is an entity? Is it a string? Do you accept a list of string entities or list of numeric ? I am not clear

Reply:
Added an Example in doc